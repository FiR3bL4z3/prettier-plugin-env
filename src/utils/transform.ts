export type TransformFn<T> = (nodes: T[]) => T[];
export type ConditionalTransformFn<T> = {
  condition: boolean;
  transform: TransformFn<T>;
};

export type Transformer<T> = TransformFn<T> | ConditionalTransformFn<T>;

export const transform = <T>(ast: T[], transformFns: Transformer<T>[]) => {
  return transformFns.reduce((acc, fn) => {
    if ("condition" in fn) {
      return fn.condition ? fn.transform(acc) : acc;
    }
    return fn(acc);
  }, ast);
};
