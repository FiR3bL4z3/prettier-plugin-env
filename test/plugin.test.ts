import { describe, it, expect } from "vitest";
import prettier from "prettier";
import plugin from "@/index";
import { readEnvFiles } from "./utils/read-env-files";

describe("Plugin Tests", async () => {
  const envFiles = await readEnvFiles();

  it.each(envFiles)(
    "formats $fileName content",
    async ({ content, fileName }) => {
      const output = await prettier.format(content, {
        parser: "dotenv",
        plugins: [plugin],
      });
      expect([fileName, output]).toMatchSnapshot();
    }
  );
});
