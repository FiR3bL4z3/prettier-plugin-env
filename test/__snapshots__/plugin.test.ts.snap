// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Plugin Tests > formats '.env.01-basic' content 1`] = `
[
  ".env.01-basic",
  "A = A
B = B",
]
`;
